plugins/by-name/nvim-surround/default.nix

```diff
@@ -0,0 +1,127 @@
+                For "static" delimiter pairs it can be a list of strings representing the value
+                of the delimiter pair.
+              '';
+              apply = v: if lib.isString v then lib.nixvim.mkRaw v else v;
+              example = [
+                "( "
+                " )"
+              ];
+            };
+            find = lib.mkOption {
+              type = with types; maybeRaw str;
+              description = ''
+                A function that returns the "parent selection" for a surrounding pair.
+                It can also be string which is interpreted as a Lua pattern that represents the
+                selection.
+              '';
+              example.__raw = ''
+                function()
+                  return M.get_selection({ motion = "a(" })
+                end
+              '';
+            };
+            delete = lib.mkOption {
+              type = with types; maybeRaw str;
+              description = ''
+                A function that returns the pair of selections to remove from the buffer when
+                deleting the surrounding pair.
+                It can also be string which is interpreted as a Lua pattern whose match groups
+                represent the left/right delimiter pair.
+              '';
+              example = "^(. ?)().-( ?.)()$";
+            };
+
+            change = lib.nixvim.mkNullOrOption (types.submodule {
+              options = {
+                target = lib.mkOption {
+                  type = with types; maybeRaw str;
+                  description = ''
+                    A function that returns the pair of selections to be replaced in the buffer
+                    when changing the surrounding pair.
+                    It can also be a string which is interpreted as a Lua pattern whose match
+                    groups represent the left/right delimiter pair.
+                  '';
+                  example = "^<([^>]*)().-([^%/]*)()>$";
+                };
+                replacement = lib.mkOption {
+                  type = types.rawLua;
+                  description = ''
+                    A function that returns the surrounding pair to replace the target
+                    selections.
+                  '';
+                  example.__raw = ''
+                    function()
+                        local result = M.get_input("Enter the function name: ")
+                        if result then
+                            return { { result }, { "" } }
+                        end
+                    end
+                  '';
+                };
+              };
+            }) "An attribute set with two keys: `target` and `replacement`.";
+          };
+        };
+      in
+      lib.nixvim.mkNullOrOption (types.attrsOf surroundType) ''
+        Associates each key with a "surround". The attribute set contains
+        the 'add', 'find', 'delete', and 'change' keys.
+
+        Default: see upstream documentation.
```

Comment:
Couldn't wee make use of the `pluginDefault` field here?

E.g. `pluginDefault = literalMD "See [upstream documentation](http://......)"`